src/App.jsx

```diff
@@ -14,45 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_LENGTH = 'sortByLength';
+const SORT_ALPHABET = 'SortAlphabetically';
+
+function sortBy(goods, sortField, isReverse) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((curent, next) => {
+      switch (sortField) {
+        case SORT_BY_LENGTH:
+          return curent.length - next.length;
+        case SORT_ALPHABET:
+          return curent.localeCompare(next);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, checkRev] = useState(false);
+  const sortedGoods = sortBy(goodsFromServer, sortField, isReverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button', 'is-info', { 'is-light': sortField !== SORT_ALPHABET },
+          )}
+          onClick={() => setSortField(SORT_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SORT_BY_LENGTH },
+          )}
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button', 'is-warning', { 'is-light': !isReverse },
+          )}
+          onClick={() => checkRev(!isReverse)}
```

Comment:
move to a handler function
```
const handleReverseClick = () => {
  setIsReversed(prev => !prev)}
}
```
```suggestion
          onClick={handleReverseClick}
```